### PR TITLE
ABLASTR: Constants

### PR DESCRIPTION
**Pull request by ax3l**
Add a new header for constants to ABLASTR. Move over constants from WarpX as its content.

Will be used in ImpactX here: https://github.com/ECP-WarpX/impactx/pull/252

### CODE REVIEW

File: Source/ablastr/Constant.H [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        //! electron mass [kg]
+        static constexpr auto m_e = static_cast<amrex::Real>( 9.1093837015e-31 );
+        //! proton mass [kg]
+        static constexpr auto m_p = static_cast<amrex::Real>( 1.67262192369e-27 );
+        //! dalton: unified atomic mass unit [kg]
+        static constexpr auto m_u = static_cast<amrex::Real>( 1.66053906660e-27 );
+
+        //! reduced Planck Constant = h / tau [J*s]
+        static constexpr auto hbar = static_cast<amrex::Real>( 1.054571817e-34 );
+        //! mu0/(4*MathConst::pi)*q_e*q_e*c/hbar
```
Comment (ax3l) [inline Source/ablastr/Constant.H:56]: @lucafedeli88 in a follow-up PR, can you potentially add equivalent docs for these constants? :)

You can also rely on the `constexpr` math now if you like, no need for the numerical values.

File: Source/ablastr/Constant.H [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        static constexpr auto m_e = 9.1093837015e-31_rt;
+        //! proton mass [kg]
+        static constexpr auto m_p = 1.67262192369e-27_rt;
+        //! dalton: unified atomic mass unit [kg]
+        static constexpr auto m_u = 1.66053906660e-27_rt;
+
+        //! reduced Planck Constant = h / tau [J*s]
+        static constexpr auto hbar = 1.054571817e-34_rt;
+        //! mu0/(4*MathConst::pi)*q_e*q_e*c/hbar
+        static constexpr auto alpha = 0.007297352573748943_rt;
```
Comment (ax3l) [inline Source/ablastr/Constant.H:61]: @lucafedeli88 in a follow-up PR, can you potentially add equivalent docs for these constants? :)

You can also rely on the `constexpr` math now if you like, no need for the numerical values.

File: Source/ablastr/constant.H [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        static constexpr auto m_e = 9.1093837015e-31_rt;
+        //! proton mass [kg]
+        static constexpr auto m_p = 1.67262192369e-27_rt;
+        //! dalton: unified atomic mass unit [kg]
+        static constexpr auto m_u = 1.66053906660e-27_rt;
+
+        //! reduced Planck Constant = h / tau [J*s]
+        static constexpr auto hbar = 1.054571817e-34_rt;
+        //! mu0/(4*MathConst::pi)*q_e*q_e*c/hbar
+        static constexpr auto alpha = 0.007297352573748943_rt;
```
Comment (ax3l) [inline Source/ablastr/constant.H:61]: @lucafedeli88 in a follow-up PR, can you potentially add equivalent docs for these constants? :)

You can also rely on the `constexpr` math now if you like, no need for the numerical values.

**Comment by ax3l** [comment]
@RemiLehe ping :)

File: Source/ablastr/constant.H (hunk 3 above)
Comment (lucafedeli88) [reply Source/ablastr/constant.H:61]: Do you mean adding the units, like `[m]` ? Most are actually dimensionless constants, except for the classical electron radius `r_e`, which is a length. I can certainly add for each of them its proper name...

File: Source/ablastr/constant.H [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    /** Mathematical constants */
+    namespace math
+    {
+        using namespace amrex::literals;
+
+        //! ratio of a circle's circumference to its diameter
+        static constexpr amrex::Real pi = 3.14159265358979323846_rt;
+
+        //! https://tauday.com/tau-manifesto
+        static constexpr amrex::Real tau = 2.0 * 3.14159265358979323846_rt;
```
Comment (lucafedeli88) [inline Source/ablastr/constant.H:26]: Thanks for the link to the tau-manifesto !

I am wondering if writing something like that could be better for readability...
```suggestion
        static constexpr amrex::Real tau = 2.0 * pi;
```

File: Source/ablastr/constant.H [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        //! electron mass [kg]
+        static constexpr auto m_e = 9.1093837015e-31_rt;
+        //! proton mass [kg]
+        static constexpr auto m_p = 1.67262192369e-27_rt;
+        //! dalton: unified atomic mass unit [kg]
+        static constexpr auto m_u = 1.66053906660e-27_rt;
+
+        //! reduced Planck Constant = h / tau [J*s]
+        static constexpr auto hbar = 1.054571817e-34_rt;
+        //! mu0/(4*MathConst::pi)*q_e*q_e*c/hbar
```
Comment (lucafedeli88) [inline Source/ablastr/constant.H:60]:
```suggestion
        //! fine-structure constant =  mu0/(4*MathConst::pi)*q_e*q_e*c/hbar [dimensionless]
```

File: Source/ablastr/constant.H (hunk 3 above)
Comment (lucafedeli88) [reply Source/ablastr/constant.H:61]: Sure! I've done that for some of the constants in my comments...

File: Source/ablastr/constant.H [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        static constexpr auto alpha = 0.007297352573748943_rt;
+        //! 1./(4*MathConst::pi*ep0) * q_e*q_e/(m_e*c*c)
```
Comment (lucafedeli88) [inline Source/ablastr/constant.H:62]:
```suggestion
        //!  classical electron radius = 1./(4*MathConst::pi*ep0) * q_e*q_e/(m_e*c*c) [m]
```

**Review by lucafedeli88 — APPROVED** [review]
Thanks for this PR! I've left a couple of comments

**Comment by ax3l** [comment]
Oops, I already had that on auto-merge and the `approve` merged it accordingly :sweat_smile: 

Do you want to push this in a follow-up? :)
Update: #3410

**Pull request merged**